src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -65,6 +65,17 @@
  */
 public class CheckstyleAntTask extends Task {
 
+    /**
+     * Shared DirectoryScanner instance used for XML formatting operations.
+     *
+     * <p>This serves as a lightweight alternative to creating multiple scanner instances,
+     * particularly for simple XML formatting tasks where thread safety isn't required.
+     * The scanner is intentionally kept package-private for internal use only.</p>
+     *
+     * <p>Note: This is not thread-safe for concurrent operations. For thread-safe usage,
+     * create separate scanner instances.</p>
+     */
+    private static final DirectoryScanner SCANNER = new DirectoryScanner();
```

Comment:
`spotbugs:`

```
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.scanPath(Path, int) allocates an object that is used in a constant way in a loop [com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask] At CheckstyleAntTask.java:[line 540] PCAIL_POSSIBLE_CONSTANT_ALLOCATION_IN_LOOP
```

Reply:
creationof this bean is ease/light , better to avoid extra  "state" (field in class) that we will need to setup by ".set"

Reply:
done